xarray/coding/cftime_offsets.py

```diff
@@ -1218,6 +1218,9 @@ def date_range(
     """
     from xarray.coding.times import _is_standard_calendar
 
+    if freq is None and any(arg is None for arg in [periods, start, end]):
+        freq = "D"
```

Comment:
Sorry on second thought could you move this logic into `cftime_range` (defined a little farther up in this file)?  That way we can harmonize the default arguments across `pandas.date_range`, `xarray.cftime_range`, and `xarray.date_range`.